### PR TITLE
Read calibration options from config file

### PR DESCRIPTION
**Pull request by dfinstad**

### CODE REVIEW

**Comment by dfinstad** [comment]
@cmbiwer let me know what you think of the general idea of what I'm doing here. I'm imagining I'll read in the transfer function paths and the reference values `fc0`, `fs0`, `qinv0` from the `[calibration]` section in the config file, and then sampling will be done in `delta_fs`, `delta_fc`, and `delta_qinv` (in addition to the kappas)

**Comment by cmbiwer** [comment]
@dfinstad Yes, this is what I was hoping for. It looks like your still pushing commits. Let me know when you are done.

**Comment by dfinstad** [comment]
@cmbiwer Okay, finished.

File: pycbc/calibration/__init__.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    section : {"calibration", string}
+        Section name from which to retrieve the model.
+
+    Returns
+    -------
+    instance
+        An instance of the calibration model class.
+    """
+    name = cp.get_opt_tag(section, "name")
+    model_args = models[name].from_config(cp, ifo)
```
Comment (cmbiwer) [inline pycbc/calibration/__init__.py]: Ah wait, here, just have it return the instance of ``Recalibrate`` not the args.

So the lines below here just put those inside ``from_config``.

File: pycbc/calibration/recalibrate.py [hunk 2]
```diff
@@ -242,3 +251,59 @@ def adjust_strain(self, fs=None, qinv=None, delta_fc=None, kappa_c=1.0,
...
+            tf_path = cp.get_opt_tag("calibration", tag)
+            tfs.append(self.tf_from_file(tf_path))
+        a_tst0 = tfs[0][:, 1]
+        a_pu0 = tfs[1][:, 1]
+        c0 = tfs[2][:, 1]
+        d0 = tfs[3][:, 1]
+        freq = tfs[0][:, 0]
+
+        # read fc0, fs0, and qinv0
+        fc0 = cp.get_opt_tag("calibration", '-'.join([ifo, "fc0"]))
```
Comment (cmbiwer) [inline pycbc/calibration/recalibrate.py]: Also you had ``section`` in the function above but hardcoded ``"calibration"`` here. Pass ``section`` to this function and use it here.

**Comment by dfinstad** [comment]
@cmbiwer I think I got what you meant, but let me know if I've returned the class instance correctly since that was new to me.

**Comment by cmbiwer** [comment]
Make it a class method:
```
@classmethod
def from_config(cls, cp, strain, ifo, section):
```

Then instead do:
```
 return cls(strain, freq=freq, fc0=fc0, c0=c0, d0=d0,
```

**Comment by dfinstad** [comment]
I looked at doing that, but I then I wouldn't be able to make the call to `self.tf_from_file` would I? Is there another way to make that call, or would I need to move that function somewhere else?

**Comment by cmbiwer** [comment]
You should be able to do ``tf = cls.tf_from_file(path)``.

**Comment by dfinstad** [comment]
Ah, of course, thanks. Should be fixed now.

**Comment by cmbiwer** [comment]
Cool. Once it passes the checks this can be merged.

**Comment by cdcapano** [comment]
@cmbiwer @dfinstad One thing I noticed here and in #1893 (should of commented there, sorry), is that you're saving the strain as an attribute of the Recalibrate class. Why not have `strain` be an argument that is passed to the `adjust_strain` function instead? Then you don't need to pass a strain argument to `read_model_from_config`. When we use this in PE, we'll be applying the recalibration to the template waveforms, not the strain data. That means the strain this is being applied to will change on every single iteration. It'll be a bit clearer (and safer) if the strain that `adjust_strain` is being applied to is an input argument, rather than having to change an attribute of the class prior to calling the function.

**Comment by dfinstad** [comment]
@cdcapano It was done this way on the suggestion of @cmbiwer, and my (tacit) understanding has been that the recalibration would be done on the strain data, not the template waveforms, since this is where the calibration errors would be present. In that case passing the strain once and being done with it would make sense.

**Comment by cmbiwer** [comment]
I don't think its "cleaner and safer", I think its a matter of preference. @dfinstad  just switch back to the older version you had that passed ``strain`` to ``adjust_strain``. Sorry about making you undo changes.

**Comment by dfinstad** [comment]
No problem, it's done. As far as investigating calibration errors is concerned, is there a difference between adjusting the strain or the templates here?

**Pull request merged**